### PR TITLE
Merging in my branch after megamerge with Andreas.

### PR DESCRIPTION
**Pull request by DaRavenox**
See commits for changes history. Mainly creating and removing calendars.

### CODE REVIEW

**Review by dskeppstedt — COMMENTED** [review]
Will hand this one over to @AndreasBrommund

File: dev/js/authenticate.js [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
 			document.getElementById('customBtn'), //what document element is the click attached to
 			{}, //OPTIONS
 			function() {auth2.grantOfflineAccess().then(function(resp) {
 				console.log(resp.code);
 				startAppHtmlElementManager.swapVisibility(0,1,"hidden","visible");
-				var dataToBeSent = {auth_code: resp.code}
-				startAppHttpCommunicator.sendToServer("http://127.0.0.1:5000/login",dataToBeSent);
+				let dataToBeSent = {auth_code: resp.code}
+				startAppHttpCommunicator.communicateWithServer("http://127.0.0.1:5000/login",dataToBeSent,"POST").then(function(resp){
+					if(resp == 1){
```
Comment (AndreasBrommund) [inline dev/js/authenticate.js:18]: Is it possible to use ===

File: dev/js/authenticate.js [hunk 2]
```diff
@@ -26,4 +37,45 @@ var startApp = function() {
 			startAppHtmlElementManager.swapVisibility(0,1,"visible","hidden")
 		});
 	});
-};
\ No newline at end of file
+
+	/*TEST CODE FOR CALENDAR CREATION
```
Comment (AndreasBrommund) [inline dev/js/authenticate.js:41]: Can u remove this comment?

File: dev/js/authenticate.js [hunk 3]
```diff
@@ -26,4 +37,45 @@ var startApp = function() {
...
+      					'calendarId': idToBeDeleted.id
+      				});     
+    					request2.execute(function(resp) {
+      						console.log("iTime calendar deleted")
+    				});
+    			}
+    			else{
+    				console.log("No iTime calendar exists")
+    			}
+    		})
```
Comment (AndreasBrommund) [inline dev/js/authenticate.js:78]: Indent correct

**Review by AndreasBrommund — CHANGES_REQUESTED** [review]

File: dev/js/authenticate.js (hunk 1 above)
Comment (AndreasBrommund) [reply dev/js/authenticate.js:18]: Maybe use trim, if it's a problem with newline.

https://www.w3schools.com/jsref/jsref_trim_string.asp

@DaRavenox

**Review by AndreasBrommund — APPROVED** [review]

**Pull request merged**